### PR TITLE
L2-819: Leave Community Fund functionality

### PR DESCRIPTION
**Pull request by lmuntaner**
# Motivation

Add functionality for a neuron to leave the community fund.

# Changes

* Updated candid files and derived.
* Add LeaveCommunityFund operation type.
* Add "leaveCommunityFund" method to governance canister.

# Tests

* Add test for new method in the governance canister.

### CODE REVIEW

**Comment by lmuntaner** [comment]
@peterpeterparker please review.

There are a lot of files changes, but most of them are because I updated candid. I guess it makes sense that when we update one candid files we also update the other ones, right? So that we make sure they are still up to date.

There are many changes due to this ActorMethod when compiling the JS files from the candid ones. Was this something you added?

**Comment by lmuntaner** [comment]
I reverted the candid changes for the sns-js package. That probably needs some work if we want to update them and therefore better to be done in another PR.

**Review by peterpeterparker — CHANGES_REQUESTED** [review]
Just rejected it because it would break. You need to generate the candid file with the exact `didc` version as the one used in the CI.

**Comment by lmuntaner** [comment]
I used the `compile-idl-js`, isn't that the way?

Which version should I use? The pipeline passed...

**Comment by peterpeterparker** [comment]
👉 https://github.com/dfinity/ic-js/blob/42064a5a780e2b1c62b45ef9ade606fb509ef503/.github/workflows/checks.yml#L46

**Comment by lmuntaner** [comment]
@peterpeterparker thanks, updated the PR after compiling with the same didc version.

But should the CI have failed? because it didn't... worth checking that I guess then.

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,7 @@
 # 0.x.y (2022-aa-bb)
 
+- Add "leaveCommunityFund" functionality to governance canister.
```
Comment (peterpeterparker) [inline CHANGELOG.md:3]: can you add this under "### Features" or any other related type?

File: packages/nns/src/governance.canister.ts [hunk 2]
```diff
@@ -341,6 +342,20 @@ export class GovernanceCanister {
     });
   };
 
+  /**
+   * Neuron joins the community fund
```
Comment (peterpeterparker) [inline packages/nns/src/governance.canister.ts:346]: comment does not match

**Comment by peterpeterparker** [comment]
Thanks for the didc. Yes the ci would have failed because it would have detected differences but since you removed the sns changes from your pr I guess it's why ultimately it worked out because there werent' any other changes related to the didc version.

File: CHANGELOG.md (hunk 1 above)
Comment (lmuntaner) [reply CHANGELOG.md:3]: Ah, I didn't realize those were for the specific release. Sure.

File: packages/nns/src/governance.canister.ts (hunk 2 above)
Comment (lmuntaner) [reply packages/nns/src/governance.canister.ts:346]: thanks!

File: CHANGELOG.md (hunk 1 above)
Comment (lmuntaner) [reply CHANGELOG.md]: done

**Comment by lmuntaner** [comment]
@peterpeterparker ready for another review

**Review by peterpeterparker — APPROVED** [review]
Thanks

**Pull request merged**